### DISCUSSION TITLE
Phonemes Gruut language

### DISCUSSION THREAD

**Discussion by code-ascend** [coqui-ai/TTS#2474] [Ideas]
Hello

requirements:
# gruut+supported langs
gruut[de]==2.2.3

To support the German language and default english, but this library also supports many other languages. In particular, I encountered problems in the code for the Russian language, namely:

File "TTS/TTS/tts/datasets/dataset.py", line 220, in get_phonemes
assert len(out_dict["token_ids"]) > 0

AssertionError

When language support is not explicitly specified, the library simply returns empty arrays for phonemes. 

I suggest specifying all supported languages in the dependencies or adding information about this in the documentation.

**Comment by ayyaz-azeem** [comment]
using code that was found in one of the internal files

from TTS.tts.utils.text.phonemizers.gruut_wrapper import Gruut
GRUUT_LANGS = list(Gruut.supported_languages())
print(len(GRUUT_LANGS))

OUTPUT: 16-> meaning 16 languages supported

then i printed all of those 16 languages that were
GRUUT_LANGS

OUTPUT:
['sv-se',
 'ar',
 'sw',
 'pt',
 'cs-cz',
 'nl',
 'de-de',
 'en-gb',
 'fr-fr',
 'it-it',
 'ru-ru',
 'zh-cn',
 'lb',
 'es-es',
 'fa',
 'en-us']
 
hope that's was what you were looking for!

**Comment by r-dh** [comment]
I want to add that it's not because a language is claimed to be supported by Gruut, that you'll get good results with it. My experience with Dutch was that I got completely mangled output, as if someone synthesized random keyboard smashing.
